### ISSUE TITLE
Reading Json File with object in it 

### ISSUE THREAD

**Issue by aersam** [jorgecarleitao/arrow2#1389]
Port of https://github.com/pola-rs/polars/issues/6745 

I have a Json like this:

```json
{
    "value": [
        { "col1": 1, "col2": true },
        { "col1": 3, "col2": false}
   ]
}
```

Reading gives:
`RuntimeError: BindingsError: "ArrowError(NotYetImplemented(\"read an Array from a non-Array JSON\"))"`

Any chance to implement that?

**Comment by jorgecarleitao** [comment]
Yes. Just to make sure I understand, in the case of

```json
{
    "a": [1, 2],
    "b": [1, 2, 3],
}
```

should we error? I.e. we require that every key has the same number of elements?

**Comment by aersam** [comment]
No. This should produce one row with cols a and b and and both are of type array[int]

**Comment by aersam** [comment]
Users will probably need to call „explode“ - if using polars - to produce something better

**Comment by aersam** [comment]
Alternatively one could also allow passing a kind of Json Path to read Json, something like $.value which would then just read that part of the json
